hibernate-core/src/main/java/org/hibernate/dialect/pagination/AbstractLimitHandler.java

```diff
@@ -8,16 +8,25 @@
 
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.hibernate.engine.spi.RowSelection;
 
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+import static java.util.regex.Pattern.compile;
+
 /**
  * Default implementation of {@link LimitHandler} interface. 
  *
  * @author Lukasz Antoniak (lukasz dot antoniak at gmail dot com)
  */
 public abstract class AbstractLimitHandler implements LimitHandler {
 
+	private static final Pattern FOR_UPDATE_PATTERN =
+			compile("\\s+for\\s+update\\b|\\s*(;|$)", CASE_INSENSITIVE);
```

Comment:
Will the above pattern matches any string for any string will match the `\\s*(;|$)`? Seems the following is correct:
```
compile("\\s+for\\s+update\b", CASE_INSENSITIVE);
```
Also see my other comment regarding the robustness of such string pattern matching.